frontend/public/components/namespace-dropdown.tsx

```diff
@@ -0,0 +1,327 @@
+      (!checkIsFavorite || isFavorite(option))
+    );
+  };
+  const noResults = (
+    <>
+      <Divider />
+      <EmptyState>
+        <Title size="md" headingLevel="h4">
+          {isProjects ? t('public~No projects found') : t('public~No namespaces found')}
+        </Title>
+        <EmptyStateBody>{t('public~No results match the filter criteria.')}</EmptyStateBody>
+        <EmptyStatePrimary>
+          <Button
+            variant="link"
+            onClick={(event) => {
+              event.preventDefault();
+              event.stopPropagation();
+              setFilterText('');
+              filterRef.current?.focus();
+            }}
+            className="co-namespace-selector__clear-filters"
+          >
+            {t('Clear filters')}
+          </Button>
+        </EmptyStatePrimary>
+      </EmptyState>
+    </>
+  );
+
+  const filteredOptions = (favorites?: boolean) =>
+    options.filter((option) => isOptionShown(option, favorites));
+
+  const menuGroup = (favorites?: boolean) => {
+    const label = favorites ? t('public~Favorites') : t('public~Projects');
+
+    if (filteredOptions(favorites).length === 0) {
+      return favorites ? null : noResults;
+    }
+
+    const menuItems = filteredOptions(favorites).map((option) => {
+      return (
+        <MenuItem
+          translate="true"
+          key={option.key}
+          itemId={option.key}
+          isFavorited={isFavorite(option)}
+          isSelected={selected === option.key}
+          isActive={activeOption === option.key}
+        >
+          {option.title}
+        </MenuItem>
+      );
+    });
+    return (
+      <>
+        <Divider />
+        <MenuGroup label={label} translate="true">
+          <MenuContent translate="true">
+            <MenuList translate="true">{menuItems}</MenuList>
+          </MenuContent>
+        </MenuGroup>
+      </>
+    );
+  };
+
+  const actionList = (
+    <MenuList translate="true">
+      {actions.map((action) => {
+        return (
+          <div className="co-namespace-selector__footer-item" key={action.actionKey}>
+            <Button
+              variant="secondary"
+              onClick={(event) => {
+                setMainSelectOpen(false);
+                onSelect(event, action.actionKey);
+              }}
+              className="co-namespace-selector__action-button"
+            >
+              {action.actionTitle}
+            </Button>
+          </div>
+        );
+      })}
+    </MenuList>
+  );
+
+  const footer =
+    actions?.length > 0 ? (
+      <Menu
+        onSelect={(event: React.MouseEvent, itemId: string) => {
+          setMainSelectOpen(false);
+          onSelect(event, itemId);
+        }}
+        onGetMenuHeight={(menuId, height) => setFooterHeight(height)}
+        className="co-namespace-selector__footer"
+      >
+        {actionList}
+      </Menu>
+    ) : null;
+
+  const NamespaceChildSelect = (
```

Comment:
nit, confusing name. Maybe just `NamespaceMenu`?

Reply:
Another artifact from the previous approach of nesting Select components.  Renamed on next commit.